include/engine/datafacade/datafacade_base.hpp

```diff
@@ -70,20 +71,28 @@ class BaseDataFacade
     virtual util::Coordinate GetCoordinateOfNode(const unsigned id) const = 0;
     virtual OSMNodeID GetOSMNodeIDOfNode(const unsigned id) const = 0;
 
-    virtual unsigned GetGeometryIndexForEdgeID(const unsigned id) const = 0;
+    virtual extractor::GeometryID GetGeometryIndexForEdgeID(const unsigned id) const = 0;
 
-    virtual void GetUncompressedGeometry(const EdgeID id,
-                                         std::vector<NodeID> &result_nodes) const = 0;
+    virtual void GetUncompressedForwardGeometry(const EdgeID id,
+                                                std::vector<NodeID> &result_nodes) const = 0;
```

Comment:
I don't think we ever use the existing data in `result_nodes`. I'll change this to return the vector.